src/java/org/apache/cassandra/service/accord/db/AccordUpdate.java

```diff
@@ -689,22 +695,13 @@ public long serializedBodySize(IncrementingUpdate update, int version)
         };
     }
 
-    private static Keys keysFrom(List<AbstractUpdate> updates, List<UpdatePredicate> predicates)
-    {
-        Set<Key> keys = new HashSet<>();
-        for (AbstractUpdate update : updates)
-            keys.add(update.partitionKey());
-        for (UpdatePredicate predicate : predicates)
-            keys.add(new PartitionKey(predicate.table.id, predicate.key));
-
-        return new Keys(keys);
-    }
-
     public AccordUpdate(List<AbstractUpdate> updates, List<UpdatePredicate> predicates)
```

Comment:
Modifies its inputs, which for efficiency is of course good. It's only used from `AccordTxnBuilder` so seems fine.